src/batterymonitor.cpp

```diff
@@ -77,9 +77,12 @@ void BatteryMonitor::Loop()
                     }
                 }
             #endif
-            #if BATTERY_MONITOR == BAT_EXTERNAL
+            #if ESP8266 && BATTERY_MONITOR == BAT_EXTERNAL
                 voltage = ((float)analogRead(PIN_BATTERY_LEVEL)) * batteryADCMultiplier;
             #endif
+            #if ESP32 && BATTERY_MONITOR == BAT_EXTERNAL
+                voltage = ((float)analogReadMilliVolts(PIN_BATTERY_LEVEL)) / 1000 * (BATTERY_SHIELD_RESISTANCE + BATTERY_SHIELD_R1 + BATTERY_SHIELD_R2) / BATTERY_SHIELD_R1;
```

Comment:
Why not just modify `batteryADCMultiplier`?

Reply:
From what I understand, the multiplier is correct but ESP32 ADC pins aren't linear and `analogRead` doesn't use the calibration that every ESP32 ships with, while `analogReadMillivolts` does

Reply:
Mmm I guess, I just mean basically copying the code for the multiplier seems a little silly

Reply:
I didn't see a better way to do it without also refactoring ADC behavior for ESP8266